### PR TITLE
my account template

### LINKED ISSUE #127: Dar funcionalidad a pantalla de inicio "Mi Cuenta"
- Crear endpoint para actualizar únicamente el nombre de usuario (actualmente utiiliza un endpoint que recibe todo el objeto User, ese endpoint no debe ser usado desde esta pantalla)
- Al dar click en guardar utilizar el nuevo endpoint para actualizar el nombre de usuario
- Eliminar la tabla que se muestra actualmente
- Añadir un nuevo componente con información por casa del usuario, el componente deberá incluir:
    - Calle
    - Número de casa
    - Chips asignados a la casa
    - Indicadores de bimestres pagados en el año actual (similar a la vista de circuito pero solo para la casa del componente)
    - Teléfonos registrados a la casa (por el momento solo mostrar el título "Teléfonos en caseta")

Nota: El componente para mostrar la información de la casa debe cargar la información de su casa en el onInit de ese componente, de modo que las diferentes casas se carguen de forma concurrente, mientras se cargue la información deberá mostrar el spinner cada componente

### PR DESCRIPTION
**Pull request by fakeghostl**

### CODE REVIEW

File: backend/src/main/java/org/uresti/pozarreal/service/impl/HousesServiceImpl.java [hunk 1]
```diff
@@ -134,4 +132,29 @@ public HouseByUser saveHouseByUser(HouseByUser houseByUser) {
 
         return HousesByUserMapper.entityToDto(housesByUserRepository.save(HousesByUserMapper.dtoToEntity(houseByUser)));
     }
+
+    @Override
+    public ArrayList<PaymentByConcept> getPaymentsHouse(String houseId) {
+        LocalDate startOfYear = LocalDate.now().withYear(LocalDate.now().getYear()).withDayOfYear(1);
```
Comment (iuresti) [inline backend/src/main/java/org/uresti/pozarreal/service/impl/HousesServiceImpl.java]: Are you sure ".withYear(LocalDate.now().getYear())" is doing something? .. I mean, the base date is LocalDate.now()

File: backend/src/main/java/org/uresti/pozarreal/service/impl/HousesServiceImpl.java [hunk 2]
```diff
@@ -134,4 +132,29 @@ public HouseByUser saveHouseByUser(HouseByUser houseByUser) {
...
+
+        LocalDate endOfYear = LocalDate.now().withYear(LocalDate.now().getYear() + 1).withDayOfYear(1);
```
Comment (iuresti) [inline backend/src/main/java/org/uresti/pozarreal/service/impl/HousesServiceImpl.java]: Same thing than in previous line

File: backend/src/main/java/org/uresti/pozarreal/tools/PaymentTools.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import org.uresti.pozarreal.dto.House;
+import org.uresti.pozarreal.dto.PaymentByConcept;
+import org.uresti.pozarreal.model.Payment;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.uresti.pozarreal.model.PaymentSubConcept.*;
+
+public class PaymentTools {
```
Comment (iuresti) [inline backend/src/main/java/org/uresti/pozarreal/tools/PaymentTools.java:14]: I liked the separation of this responsability

File: backend/src/main/java/org/uresti/pozarreal/tools/PaymentTools.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    public static void setYearPayments(House house, List<Payment> payments) {
+        String[] twoMonthsPaymentIds = {
```
Comment (iuresti) [inline backend/src/main/java/org/uresti/pozarreal/tools/PaymentTools.java]: Is this being covered by a unit test ?

**Review by iuresti — CHANGES_REQUESTED** [review]

File: backend/src/main/java/org/uresti/pozarreal/tools/PaymentTools.java (hunk 3 above)
Comment (fakeghostl) [reply backend/src/main/java/org/uresti/pozarreal/tools/PaymentTools.java:14]: the problem is that i'm not inject FeeConfig, how can i do it?

File: backend/src/main/java/org/uresti/pozarreal/tools/PaymentTools.java (hunk 4 above)
Comment (fakeghostl) [reply backend/src/main/java/org/uresti/pozarreal/tools/PaymentTools.java]: no exactly, is like a black box

File: backend/src/main/java/org/uresti/pozarreal/service/impl/HousesServiceImpl.java (hunk 1 above)
Comment (fakeghostl) [reply backend/src/main/java/org/uresti/pozarreal/service/impl/HousesServiceImpl.java]: i tried to get current date and end with one year more

File: backend/src/main/java/org/uresti/pozarreal/tools/PaymentTools.java (hunk 3 above)
Comment (iuresti) [reply backend/src/main/java/org/uresti/pozarreal/tools/PaymentTools.java:14]: I didn't notice that .. thanks for bring that up.

In order to inject FeeConfig here you should avoid using static methods ... spring boot performs the injection when the object is created, the method setYearPayments should belong to an object.

Other available option would be pass the FeeConfig as parameter to setYearPayments and inject FeeConfig in the object calling setYearPayments

**Review by iuresti — CHANGES_REQUESTED** [review]

File: backend/src/main/java/org/uresti/pozarreal/tools/PaymentTools.java (hunk 3 above)
Comment (fakeghostl) [reply backend/src/main/java/org/uresti/pozarreal/tools/PaymentTools.java:14]: my first option was create a java bean but i think it didn't work

**Pull request merged**